### PR TITLE
bugfix/16925-stacked-area-nulls-bad-animation

### LINKED ISSUE #16925: Null data animated to lines on show/hide series
#### Expected behaviour
No lines on null data

#### Actual behaviour
Lines connecting non-null data

#### Live demo with steps to reproduce
https://jsfiddle.net/soetji/ousj7w0v/

Click on the Show/Hide Series button at the bottom

#### Product version
Highcharts,  v9.3.2

#### Affected browser(s)
Chrome, Firefox

### PR DESCRIPTION
**Pull request by hubertkozik**
Fixed #16925, stacked area series with null data produced bad animation.

---

Demos to test:
- OP demo: https://jsfiddle.net/BlackLabel/54e2apg3/
- Simplified demo: https://jsfiddle.net/BlackLabel/73v8h0xd/
- Areaspline demo: https://jsfiddle.net/BlackLabel/cntqasmj/

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206552081722239

### CODE REVIEW

**Review by KacperMadej — COMMENTED** [review]
A dedicated test (or at least a comment about which test of the existing tests should be used) would be nice. On the other hand, I checked all tests and demos with 'stack' in the name and all is ok or better now.

Some questions in code.

File: ts/Series/Spline/SplineSeries.ts [hunk 1]
```diff
@@ -254,6 +268,18 @@ class SplineSeries extends LineSeries {
...
 
+        if (hidden) {
+            return [
+                'C',
+                pick(lastPoint.plotX, 0),
+                pick(lastPoint.plotY, 0),
+                pick(lastPoint.plotX, 0),
+                pick(lastPoint.plotY, 0),
+                pick(lastPoint.plotX, 0),
+                pick(lastPoint.plotY, 0)
```
Comment (KacperMadej) [inline ts/Series/Spline/SplineSeries.ts:279]: This could be more DRY - `pick` twice instead of 6 times.

**Review by KacperMadej — APPROVED** [review]
The file size change looks like a potential problem.

**Comment by hubertkozik** [comment]
Closing as done alternative way in https://github.com/highcharts/highcharts/pull/21161.

**Pull request closed without merging**